tools/tdbg/adminDBCleanCommand.go

```diff
@@ -0,0 +1,262 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package tdbg
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+	"os"
+	"time"
+
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/persistence"
+	cassp "go.temporal.io/server/common/persistence/cassandra"
+	"go.temporal.io/server/common/persistence/nosql/nosqlplugin/cassandra/gocql"
+	"go.temporal.io/server/common/quotas"
+)
+
+type (
+	// ShardCleanReport represents the result of cleaning a single shard
+	ShardCleanReport struct {
+		ShardID         int32
+		TotalDBRequests int64
+		Handled         *ShardCleanReportHandled
+		Failure         *ShardCleanReportFailure
+	}
+
+	// ShardCleanReportHandled is the part of ShardCleanReport of executions which were read from corruption file
+	// and were attempted to be deleted
+	ShardCleanReportHandled struct {
+		TotalExecutionsCount     int64
+		SuccessfullyCleanedCount int64
+		FailedCleanedCount       int64
+	}
+
+	// ShardCleanReportFailure is the part of ShardCleanReport that indicates a failure to clean some or all
+	// of the executions found in corruption file
+	ShardCleanReportFailure struct {
+		Note    string
+		Details string
+	}
+
+	// CleanProgressReport represents the aggregate progress of the clean job.
+	// It is periodically printed to stdout
+	CleanProgressReport struct {
+		NumberOfShardsFinished     int
+		TotalExecutionsCount       int64
+		SuccessfullyCleanedCount   int64
+		FailedCleanedCount         int64
+		TotalDBRequests            int64
+		DatabaseRPS                float64
+		NumberOfShardCleanFailures int64
+		ShardsPerHour              float64
+		ExecutionsPerHour          float64
+	}
+
+	// CleanOutputDirectories are the directory paths for output of clean
+	CleanOutputDirectories struct {
+		ShardCleanReportDirectoryPath    string
+		SuccessfullyCleanedDirectoryPath string
+		FailedCleanedDirectoryPath       string
+	}
+
+	// ShardCleanOutputFiles are the files produced for a clean of a single shard
+	ShardCleanOutputFiles struct {
+		ShardCleanReportFile    *os.File
+		SuccessfullyCleanedFile *os.File
+		FailedCleanedFile       *os.File
+	}
+)
+
+func cleanShard(
```

Comment:
I think we should remove clean shard and scan shard command. @yycptt what do you think?

Reply:
I am not sure what the clean shard command do actually. But everything related to scan shard can be removed.

Reply:
@yiminc removed DB commands (except for workflow delete)